### DISCUSSION TITLE
Removal of Camera1Base.setMicrophoneMode in 2.5.2

### DISCUSSION THREAD

**Discussion by replicant1** [pedroSG94/RootEncoder#1691] [General]
Hi Pedro,

I have been calling `rtmpCamera1.setMicrophoneMode(MicrophoneMode.SYNC)` in my code for some time. But I see that the `setMicrophoneMode` method has been removed in RootEncoder `2.5.2`. Now that I am using RootEncoder `2.5.5`, what would you suggest I call instead of `setMicrophoneMode(MicrophoneMode.SYNC)`?

Thanks for your time.

Rod Bailey

**Comment by pedroSG94** [comment]
Hello,

Sorry for late response. I'm on holidays

In version 2.5.5 you can remove the method because the timestamp calculation was improved but you have 2 ways to calculate it now:
```kotlin
setTimestampMode(timestampModeVideo = TimestampMode.CLOCK, timestampModeAudio = TimestampMode.BUFFER)
```

CLOCK is used by default in both, video and audio, and BUFFER is a way to calculate the timestamp depend of the buffer information and could be better in fews cases. I recommend you test it and use the best for your case. 
You can combine it as you want. Call that method before prepareVideo/prepareAudio

**Reply by replicant1** [reply]
Hi Pedro,

Thanks for your response. I will experiment with various `setTimestampMode` arguments to see what works.

Regards,

Rod Bailey
